src/compiler/checker.ts

```diff
@@ -11161,14 +11347,20 @@ namespace ts {
                     inferFromTypes((<IndexedAccessType>source).objectType, (<IndexedAccessType>target).objectType);
                     inferFromTypes((<IndexedAccessType>source).indexType, (<IndexedAccessType>target).indexType);
                 }
+                else if (source.flags & TypeFlags.Conditional && target.flags & TypeFlags.Conditional) {

```

Comment:
Not sure I understand what you're asking.

Reply:
If source is not a conditional type, maybe we can still do some inference, similar to the way that we can infer from mapped types when the target is a homomorphic mapped type. Since we would be "running it backward" in a similar way, I thought that the key would be relating the source type to the conditional's extendsType.